lib/include/pl/patterns/pattern_array_static.hpp

```diff
@@ -23,12 +23,21 @@ namespace pl {
             auto entry = std::shared_ptr(this->m_template->clone());
             for (u64 index = 0; index < std::min<u64>(end, this->m_entryCount); index++) {
                 entry->clearFormatCache();
-                entry->setVariableName(fmt::format("[{0}]", index));
+                entry->setVariableName(this->m_template->getVariableName() + fmt::format("[{0}]", index));
                 entry->setOffset(this->getOffset() + index * this->m_template->getSize());
                 fn(index, *entry);
             }
         }
 
+        void setInlined(bool isInlined) override {
+            Inlinable::setInlined(isInlined);
+            if (isInlined) {
+                this->m_template->setVariableName(this->getVariableName());
+            } else {
+                this->m_template->setVariableName("");
```

Comment:
I wonder, if it would be better to store format in varname instead of string cat at draw time.